### PR TITLE
adding amp evaluation step test to increase coverage

### LINKED ISSUE #1790: Improve code coverage
Our project code coverage start to decrease with new features and reworked tests. We are now ~95% but were 96% recently and around ~98% the last year...

The idea is to identify the uncovered parts of code inspecting here, for example: https://codecov.io/gh/pytorch/ignite/tree/master/ignite and draft a PR with new tests.

Some of new test can be very simple and others could be very sophisticated to enter a particular code line.
Let's improve the coverage of all modules, except `distributed` for instance.

### PR DESCRIPTION
**Pull request by Moh-Yakoub**
Addresses partly #1790

Description: As part of the effort to increase cov tests, I've added few tests to test the `evaluation_Step_amp` method that wasn't covered in all tests: https://codecov.io/gh/pytorch/ignite/src/ba57568cf7d4343d22171645848c4af56a0a0ae4/ignite/engine/__init__.py

In order to do this I did refactor the autocast method to make it testable and added a test for it as well

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/engine/__init__.py [hunk 1]
```diff
@@ -499,22 +499,27 @@ def supervised_evaluation_step_amp(
...
             x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
-            with autocast(enabled=True):
+            with _autocast(enabled=True):
                 y_pred = model(x)
             return output_transform(x, y, y_pred)
 
     return evaluate_step
 
 
+def _autocast(enabled: bool) -> Any:
```
Comment (sdesrozis) [inline ignite/engine/__init__.py:514]: I don't remember where I saw this wrapper around amp but it is a good idea. Maybe to be moved in `ignite.distributed.auto.py` ?

Comment (Moh-Yakoub) [reply ignite/engine/__init__.py]: @sdesrozis Moved it and updated the tests. Let's wait to make sure all other tests passed.

Comment (Moh-Yakoub) [reply ignite/engine/__init__.py]: @sdesrozis any other comments on this PR?

File: ignite/distributed/auto.py [hunk 2]
```diff
@@ -341,3 +341,12 @@ def __init__(self, optimizer: Optimizer) -> None:
 
         def step(self, closure: Optional[Callable] = None) -> None:
             xm.optimizer_step(self.wrapped_optimizer, barrier=True)
+
+
+def autocast(enabled: bool) -> Any:
```
Comment (vfdev-5) [inline ignite/distributed/auto.py:346]: I do not quite understand why we have to introduce a public method for testing. This not a good practice, IMO. Let's remove it.

File: ignite/engine/__init__.py [hunk 3]
```diff
@@ -499,10 +500,6 @@ def supervised_evaluation_step_amp(
 
     .. versionadded:: 0.5.0
     """
-    try:
-        from torch.cuda.amp import autocast
-    except ImportError:
-        raise ImportError("Please install torch>=1.6.0 to use amp_mode='amp'.")
```
Comment (vfdev-5) [inline ignite/engine/__init__.py:505]: To test this case without `from torch.cuda.amp import autocast`, please do as here: https://github.com/pytorch/ignite/blob/f29fc24699ee43c7acb4d212a2333565ec03f3a6/tests/ignite/engine/test_create_supervised.py#L485-L487

File: ignite/engine/__init__.py (hunk 1 above)
Comment (vfdev-5) [reply ignite/engine/__init__.py]: @sdesrozis what is the purpose of introducing this more or less useless function in `ignite.distributed` ?

**Comment by Moh-Yakoub** [comment]
@vfdev-5  I've removed the method from the distribute module, let me know if you have furthur comments on that

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @Moh-Yakoub !
sorry for delay.

**Pull request merged**